tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -121,10 +121,73 @@ def tile_batch(t, multiplier, name=None):
     return nest.map_structure(lambda t_: _tile_batch(t_, multiplier), t)
 
 
+def gather_tree_from_array(t, parent_ids, sequence_length):
+  """Calculates the full beams for `TensorArray`s.
+
+  Args:
+    t: A `TensorArray` of size `max_time` that contains `Tensor`s of shape
+      `[batch_size, beam_width, s]` or `[batch_size * beam_width, s]` where
+      `s` is the depth shape.
+    parent_ids: The parent ids of shape `[max_time, batch_size, beam_width]`.
+    sequence_length: The sequence length of shape `[batch_size, beam_width]`.
+
+  Returns:
+    A `TensorArray` of the same size and type as `t` and where beams are sorted
+    in each `Tensor` according to `parent_ids`.
+  """
+  max_time = array_ops.shape(parent_ids)[0]
```

Comment:
parent_ids.shape[0].value or array_ops.shape(parent_ids)[0]
...
same for the other 2.